code/mission/missionparse.cpp

```diff
@@ -3036,7 +3036,13 @@ int parse_object(mission *pm, int flag, p_object *p_objp)
         parse_string_flag_list(p_objp->flags, Parse_object_flags, num_parse_object_flags, &unparsed);
         if (!unparsed.empty()) {
             for (size_t k = 0; k < unparsed.size(); ++k) {
-                WarningEx(LOCATION, "Unknown flag in parse object flags: %s", unparsed[k].c_str());
+				// catch typos or deprecations
+				if (!stricmp(unparsed[k].c_str(), "no-collide") || !stricmp(unparsed[k].c_str(), "no_collide")) {
+					p_objp->flags.set(Mission::Parse_Object_Flags::OF_No_collide);
```

Comment:
I didn't consider it worth a Warning because I never actually wanted to deprecate the `no-collide` term.  In fact, I would prefer to put it back to that variant because every other ship flag uses hyphens.

I decided to leave it as the underscore in order to minimize the chance of any typos happening again, but IMO it is the underscore that should be deprecated, not the hyphen.